addons/calendar/controllers/main.py

```diff
@@ -40,24 +40,27 @@ def view(self, db, token, action, id, view='calendar'):
             # Since we are in auth=none, create an env with SUPERUSER_ID
             env = Environment(cr, SUPERUSER_ID, {})
             attendee = env['calendar.attendee'].search([('access_token', '=', token)])
+            if attendee.event_id.id != int(id):
```

Comment:
Then add it in the search and do something if no result is found. Cleaner than post-search processing.